### PR TITLE
WIP Made coach dashboard available on ccx

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
- Previously ccx coach dashboard was only available on master course. Coach needs to open master course to access coach dashboard
- After this fix coach can access his dashboard from both master course and ccx

Issue https://github.com/mitocw/edx-platform/issues/33
@pdpinch @pwilkins @giocalitri

### CODE REVIEW

File: lms/djangoapps/ccx/views.py [hunk 1]
```diff
@@ -181,6 +187,11 @@ def create_ccx(request, course, ccx=None):
                 override_field_for_ccx(ccx, vertical, hidden, True)
 
     ccx_id = CCXLocator.from_course_locator(course.id, ccx.id)  # pylint: disable=no-member
+    with ccx_course(ccx_id) as course:
+        try:
+            allow_access(course, request.user, "ccx_coach")
+        except Role.DoesNotExist:
+            return HttpResponseBadRequest("Role does not exist.")
```
Comment (pdpinch) [inline lms/djangoapps/ccx/views.py]: When does this exception occur?

Comment (amir-qayyum-khan) [inline lms/djangoapps/ccx/views.py]: When coach creates ccx for the first time, I am assigning him role "ccx_coach" on the ccx.
Now he has role =  "ccx_coach" on both master course and ccx. He can access dashboard from both of them.

File: lms/djangoapps/ccx/plugins.py [hunk 2]
```diff
@@ -28,5 +31,22 @@ def is_enabled(cls, course, user=None):
...
+        course_key = course.id
+        role = CourseCcxCoachRole(course_key)
+        if role.has_user(user):
+            return True
+        elif isinstance(course_key, CCXLocator):
+            # On clicking ccx coach dashboard, somehow 'ccx_coach' role is deleted from users role cache.
+            # So here I am checking if course is ccx and use is coach then show ccx coach tab.
+            ccx_id = course_key.ccx
+            try:
+                ccx = CustomCourseForEdX.objects.get(pk=ccx_id, coach=user)
```
Comment (amir-qayyum-khan) [inline lms/djangoapps/ccx/plugins.py]: Assuming that One ccx has only one coach

@pdpinch

**Comment by amir-qayyum-khan** [comment]
Made upstream PR

**Pull request closed without merging**